core/src/retransmit_stage.rs

```diff
@@ -464,10 +466,19 @@ fn retransmit(
             .or_default() += 1;
 
         let mut retransmit_time = Measure::start("retransmit_to");
-        if !packet.meta.forward {
-            ClusterInfo::retransmit_to(&neighbors, packet, sock, true);
+        // If the node is on the critical path (i.e. the first node in each
+        // neighborhood), it should send the packet to tvu socket of its
+        // children and also tvu_forward socket of its neighbors. Otherwise it
+        // should only forward to tvu_forward socket of its children.
+        if anchor_node {
+            ClusterInfo::retransmit_to(&neighbors, packet, sock, /*forward socket=*/ true);
         }
-        ClusterInfo::retransmit_to(&children, packet, sock, packet.meta.forward);
+        ClusterInfo::retransmit_to(
+            &children,
+            packet,
+            sock,
+            !anchor_node, // send to forward socket!
+        );
```

Comment:
nit: For clarity, how about extracting out `should_send_to_forward_socket = !anchor_node`, and using that for both calls to `retransmit_to`

Reply:
The part which decides to send to forward socket or tvu socket is effectively redundant going forward. It only impacts that `retransmit_tree_mismatch` metric, otherwise is not anywhere in retransmit logic.

So will probably be deleted soon, and I am leaning on brevity here. I think the comments already describe the logic.